setup-homebrew/main.sh

```diff
@@ -22,28 +17,44 @@ else
     sudo mkdir -p bin etc include lib opt sbin share var/homebrew/linked Cellar
     sudo ln -sf ../Homebrew/bin/brew "$HOMEBREW_PREFIX/bin/"
     cd -
+
+    export PATH="$HOMEBREW_PREFIX/bin:$PATH"
 fi
 
-export PATH="$HOMEBREW_PREFIX/bin:$PATH"
+HOMEBREW_PREFIX="$(brew --prefix)"
+HOMEBREW_REPOSITORY="$(brew --repo)"
+HOMEBREW_CORE_REPOSITORY="$(brew --repo homebrew/core)"
+HOMEBREW_TAP_REPOSITORY="$(brew --repo "$GITHUB_REPOSITORY")"
+
 echo "::add-path::$HOMEBREW_PREFIX/bin"
 
+brew update-reset "$HOMEBREW_REPOSITORY" "$HOMEBREW_CORE_REPOSITORY"
+
 GEMS_PATH="$HOMEBREW_REPOSITORY/Library/Homebrew/vendor/bundle/ruby/"
+GEMS_HASH="$(shasum -a 256 "$HOMEBREW_REPOSITORY/Library/Homebrew/Gemfile.lock" | cut -f1 -d' ')"
+
 echo "::set-output name=gems-path::$GEMS_PATH"
-GEMS_HASH=$(shasum -a 256 "$HOMEBREW_REPOSITORY/Library/Homebrew/Gemfile.lock" | cut -f1 -d' ')
 echo "::set-output name=gems-hash::$GEMS_HASH"
 
-case "$GITHUB_REPOSITORY" in
-    */brew|*/*brew-core)
-        cd "$HOMEBREW_CORE_REPOSITORY"
-        rm -rf "$GITHUB_WORKSPACE"
-        ln -s "$HOMEBREW_CORE_REPOSITORY" "$GITHUB_WORKSPACE"
-        git fetch origin "$GITHUB_SHA"
-        git checkout --force -B master FETCH_HEAD
-        cd -
-    ;;
-esac
-
-if [ "$RUNNER_OS" = "Linux" ]; then
+# brew
+if [[ "$GITHUB_REPOSITORY" =~ ^.+/brew$ ]]; then
+    rm -rf "$GITHUB_WORKSPACE"
+    ln -s "$HOMEBREW_REPOSITORY" "$GITHUB_WORKSPACE"
+    git fetch --tags origin "$GITHUB_SHA"
+    git checkout --force -B master FETCH_HEAD
+# core taps
+elif [[ "$GITHUB_REPOSITORY" =~ ^.+/(home|linux)brew-core$ ]]; then
+    rm -rf "$GITHUB_WORKSPACE"
+    ln -s "$HOMEBREW_CORE_REPOSITORY" "$GITHUB_WORKSPACE"
+    git fetch origin "$GITHUB_SHA"
+    git checkout --force -B master FETCH_HEAD
+# third-party taps
+elif [[ "$GITHUB_REPOSITORY" =~ ^.+/homebrew-.+$ ]]; then
+    mkdir -p "$(dirname "$HOMEBREW_TAP_REPOSITORY")"
+    ln -s "$GITHUB_WORKSPACE" "$HOMEBREW_TAP_REPOSITORY"
```

Comment:
Might want to still delete this in case it's not around?

Reply:
https://github.com/Homebrew/actions/commit/067bcc9c26756746fccc540638150334a38e220a